geomstats/geometry/embedded_manifold.py

```diff
@@ -20,12 +22,67 @@ class EmbeddedManifold(Manifold):
         Optional, default: 'intrinsic'.
     """
 
-    def __init__(self, dim, embedding_manifold, default_point_type='vector',
+    def __init__(self, dim, embedding_manifold, submersion, tangent_submersion,
+                 default_point_type='vector',
                  default_coords_type='intrinsic', **kwargs):
         super(EmbeddedManifold, self).__init__(
             dim=dim, default_point_type=default_point_type,
             default_coords_type=default_coords_type, **kwargs)
         self.embedding_manifold = embedding_manifold
+        self.embedding_metric = embedding_manifold.metric
+        self.submersion = submersion
+        self.tangent_submersion = tangent_submersion
+
+    def belongs(self, point, atol=gs.atol):
+        """Evaluate if a point belongs to the manifold.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point to evaluate.
+        atol : float
+            Absolute tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the manifold.
+        """
+        belongs = self.embedding_manifold.belongs(point, atol)
+        constraint = gs.isclose(self.submersion(point), 0., atol=atol)
```

Comment:
Yes, I made this implicit assumption, but in fact, you're right, we could not make it and have a second argument specifying the singleton of which the manifold is the preimage of. This would solve the shape problem in isclose

Reply:
👍